docs/stories/bitcoin-integration.md

```diff
@@ -0,0 +1,109 @@
+The motivation here is to be able to minimise the changes that we need to make to `rust-lightning`.
+Effectively, the library will not need to understand the DLC protocol.
+It will just need to support the creation of outputs with _arbitrary scripts_.
+
+By default, a Lightning commitment transaction can contain:
+
+- Simple outputs that pay directly to Alice or Bob.
+- Revocable outputs that pay to Alice or Bob after a timelock expires.
+- Revocable HTLCs to route payments.
+
+This means that a Lightning commitment transaction expects its outputs to be of a particular kind.
+With our extensions, the commitment transaction should be able to host a new kind of _revocable_ output with arbitrary spend conditions.
+In our current design, the arbitrary spend conditions are external to the library and Lightning is oblivious to them; but the output should be revocable using the same mechanism and keys that are used for the other outputs in the transaction.
+
+![](../assets/add-custom-output-ln-protocol.png)
+
+Contrary to the other types of output, the coins that make up the new custom output can come from _both_ parties.
+This is a necessary requirement for DLCs and other interesting protocols, where both parties lock up coins in a shared output.
+
+The 10101 wallet application that uses will therefore have to handle all the DLC-specific transaction building and signing, using the shared custom output as the DLC.
+
+### Settling your CFD into the Lightning channel
+
+One huge advantage of the 10101 approach to CFD trading versus what we previously built with ItchySats is that the CFD can be settled directly back into a Lightning channel!
+This means that the off-chain state doesn't have to be committed on-chain and that the coins are readily available to make payments or do whatever else 10101 lets you do in the future.
+
+![](../assets/remove-custom-output-ln-protocol.png)
+
+This protocol is analogous to how a HTLC would be removed from a commitment transaction after a payment is claimed.
+Except for the fact that the coins in the custom output can be _split up_ between the two parties.
+
+### Force closing a channel containing a custom output
+
+There should not be anything inherently different about force closing a Lightning channel containing a custom output.
+The commitment transaction will be published on-chain and all the outputs will have different spend conditions.
+
+The effect on the custom output will be different, as the `rust-lightning` library will not ensure that the output is split among the two parties Bitcoin wallets.
+It will be the responsibility of the application built on top of `rust-lightning` to spend the custom output.
+For the 10101 wallet that will mean publishing a CET according to what the oracle attests.
+
+If you want to learn more about how a CFD protocol using DLCs works in the context of ItchySats, earlier in the year we wrote a [blog post](https://comit.network/blog/2022/01/11/cfd-protocol-explained/) about it.
```

Comment:
```suggestion
Settling a force-close scenario on layer 1 is in line with our original DLC protocol as described in [this blog post](https://comit.network/blog/2022/01/11/cfd-protocol-explained/). Since force-close scenarios are worst-case fallback scenarios it is feasible to handle them on chain rather than within the channel.
```

Not super happy about my wording - I'm trying to add some context to make it better understandable :)

Reply:
I've used some of this, thank you!